### DISCUSSION TITLE
error : "could not find jar:file ......."

### DISCUSSION THREAD

**Discussion by Pascr001** [eclipse-platform/.github#159] [Q&A]
## Summary
Hi all,
I would like to install a new (SAP) ABAP Plugin in Eclipse (Ex. ABAP Favorites) by "local" mode . But Unfortunatelly I can't because a blocking error !
(I do it by local mode ; otherwise i have proxy error on my customer office)
## Steps to reproduce
-       (menu) Help / Install new software
- (popup "install") button "ADD"
- (popup "add repository") button "Archive"
- select the zip downloaded in github
-    @ location = jar:file:/C:/Users/XXXXX/Downloads/ABAPFavorites-master.zip!/
-  clic on "Add"
- @@@ direct error message : **could not find jar:file:/C:/Users/XXXXX/Downloads/ABAPFavorites-master.zip!/**
- if I try on : location = jar:file://C:/Users/XXXXX/Downloads/ABAPFavorites-master.zip!/
- clic on "Add"
- @@@ error : **Unable to read repository at jar:file://C:/Users/XXXXX/Downloads/ABAPFavorites-master.zip!/content.xml.**

Any idea on how to solve my problem please ?
Thks,
Pascr001

**Comment by laeubi** [comment]
Please try to extract the zip file and use that file location directly.

**Reply by Pascr001** [reply]
I tried like this :
(menu) hekp / install new software / (popup "install") button ADD / (popup "add repository") button Local
here i select the unziped directory "ABAPFavorites-master" + select forlder
@ result Location : "file:/C:/Users/XXXXX/Downloads/ABAPFavorites-master/"

then, (popup "add repository") button Add

Error Could not find file:/C:/Users/XXXXX/Downloads/ABAPFavorites-master/

if I use this location:
file://C:/Users/XXXXX/Downloads/ABAPFavorites-master/
I got the error : 
![image](https://github.com/eclipse-platform/.github/assets/119660395/2397091c-1eb8-4b88-9b4f-710ebd7e8938)

**Reply by laeubi** [reply]
Then something is wrong with the zipfile but its hard to guess without knowing the actual content...

**Reply by Pascr001** [reply]
I tried with another plugin/zip (cucumber-eclipse-main) and the same issue :(((

**Comment by jonahgraham** [comment]
Did you download the "master" branch of https://github.com/fidley/ABAPFavorites?

if so, you downloaded the source files of the bundles, see the [readme](https://github.com/fidley/ABAPFavorites#readme) of that project which has the update site listed:

> Update site - https://fidley.github.io/ADTPluginsUpdateSite/UpdateSite/

Or perhaps install from the marketplace https://marketplace.eclipse.org/content/abap-favorites

**Reply by Pascr001** [reply]
yes, I downloaded the file "ABAPFavorites-master.zip" ; so I suppose it's "master" as you say ?!

I don't understand what you say in your 2nd sentence ??

for market : it's not possible because the proxy of my customer block us
(I before tried like this but no way. I even ask to the support, but proxy rule can be changed ;-(   )

**Reply by jonahgraham** [reply]
The distinction is source vs binary. You can't "install" the source files. You must install from the built versions.

**Reply by Pascr001** [reply]
how do you do to install built version ?
inside the unziped i have this : 
![image](https://github.com/eclipse-platform/.github/assets/119660395/b1dafc1c-236f-402f-bdf6-03da2d9b7912)

I tried to use directly the dir. "ABAP Favorites" ; but the same :((

**Reply by jonahgraham** [reply]
That is the source - please add https://fidley.github.io/ADTPluginsUpdateSite/UpdateSite/ as the update site and install from there.

If you cannot use that in your corporate environment, please ask  https://github.com/fidley/ABAPFavorites to publish a zip file you can use.

**Reply by Pascr001** [reply]
Hi,
I guess it's the link I used ... 

But back to my initial issue, I tried at home ; so WITHOUT VPN of my customer ; passing via the "Eclipse Market" ... and it was OK.
so my initial needs was solved ... but I still don't understand why it's not working in local mode ???!
thks for your help anyway.
